deepchem/data/data_loader.py

```diff
@@ -1692,7 +1692,7 @@ def shard_generator():
             entries = self._get_shards(inputs)
             for i, shard in enumerate(entries):
                 X = np.array(self._featurize_shard(shard))
-                y = np.array([0])
+                y = X[0].get_true_val()
```

Comment:
I think you want `y = [X[j].get_true_val() for j in len(shard)]`. Let's discuss more offline

Reply:
Added unit test to make sure true values are correct in line 42 test_dftxc.py

Reply:
Documenting offline discussion, each shard is an entry object since the entry can contain multiple systems.